cerbero/build/source.py

```diff
@@ -241,25 +241,27 @@ def __init__(self):
 
     def extract(self):
         if os.path.exists(self.build_dir):
-            # fix read-only permissions
-            if self.config.platform == Platform.WINDOWS:
-                shell.call('chmod -R +w .git/', self.build_dir, fail=False)
-            try:
-                commit_hash = git.get_hash(self.repo_dir, self.commit)
-                checkout_hash = git.get_hash(self.build_dir, 'HEAD')
-                if commit_hash == checkout_hash:
-                    return False
-            except Exception:
-                pass
-            shutil.rmtree(self.build_dir)
+            if len(self.patches) > 0:
```

Comment:
It's something we should have been doing for a long time, I would remove the directory even of there are no  patches just to be consistent